cms/src/main/java/org/onehippo/taxonomy/plugin/TaxonomySingleBrowser.java

```diff
@@ -0,0 +1,137 @@
+        super(id, model, taxonomyModel, preferredLocale);
+    }
+
+    public TaxonomySingleBrowser(String id, IModel<Classification> model, TaxonomyModel taxonomyModel, Locale preferredLocale, boolean detailsReadOnly, ITreeNodeIconProvider iconProvider, boolean multiple) {
+        super(id, model, taxonomyModel, preferredLocale, detailsReadOnly, iconProvider);
+
+        this.multiple = multiple;
+        //* Need to store this for our own access (it's private in the ancestor class)
+        this.parameterisedDetailsReadOnly = detailsReadOnly;
+    }
+
+    /**
+     * By default, the taxonomy picker allows multiple selections, but we can control
+     * that and we need to record the preference rather than assume it is always multi-select
+     */
+    public boolean getMultiple() {
+        return this.multiple;
+    }
+
+    @Override
+    protected Details newDetails(String id, CategoryModel model) {
+        return new ParameterisedDetails(id, model);
+    }
+
+    protected List<String> getKeys() {
+        return getModelObject().getKeys();
+    }
+
+    private boolean isCanonised() {
+        return getModelObject().isCanonised();
+    }
+
+    private void setCanonicalKey(String key) {
+        getModelObject().setCanonical(key);
+    }
+
+    /**
+     * Our own implementation of the {@link TaxonomyBrowser.Details} class which manages teh display of
+     * termas as they are selected in the TreeView on the left of the picker
+     */
+    protected class ParameterisedDetails extends TaxonomyBrowser.Details {
+        /**
+         * In this constructor, we're going to replace the default addition operations so that
+         * they are aware of the multiple selection status
+         * @param id
+         * @param model
+         */
+        public ParameterisedDetails(String id, CategoryModel model) {
+            super(id, model);
+
+            this.addOrReplace(new AjaxLink<Category>(ADD, model) {
+                @Override
+                public void onClick(AjaxRequestTarget target) {
+                    String key = getModelObject().getKey();
+                    List<String> keys = getKeys();
+                    keys.add(key);
+                    if (keys.size() == 1 && isCanonised()) {
+                        setCanonicalKey(key);
+                    }
+                    TaxonomySingleBrowser.this.modelChanged();
+                    target.add(container);
+                }
+
+                @Override
+                /**
+                 * The change to default behaviour is in here - the multiple check
+                 */
+                public boolean isVisible() {
+                    String key = getModelObject().getKey();
+                    boolean isMulti = TaxonomySingleBrowser.this.getMultiple();
+                    return !parameterisedDetailsReadOnly && !getKeys().contains(key) &&
+                            ((!isMulti && getKeys().size() == 0) || isMulti);
```

Comment:
Just wondering if we can simply it like `(isMulti || getKeys().size() == 0);`

Reply:
Y, will do. Good spot !